WordPress/gutenberg
# Add aria-describedby if image has no alt but has caption

## What?

If an image block has no alt text but has a caption, use the caption to describe the image.
If the image is linked but has no alt text, use the caption to describe the link.

Fixes  https://github.com/WordPress/gutenberg/issues/42057
## Why?

To add context to the image and link when alt text is missing.

## How?

Uses the image id to create an id that is added the caption. This could be improved since it is not truly unique.
The id is then used with the aria-describedby that is placed on the href and the image.

**Alternatives:**
- Add the caption to the alt attribute when the alt is missing-
- Use aria-labeledby to create an accessible name. However, the described by is more suitable for longer texts, and we don't know how much content is in the caption.

## Testing Instructions

1. Add an image block that does not have an alt text.
2. Add a caption to it and duplicate the block.
3. Please test both linked and unlink images.

View the source to confirm that:
- An id is added to the figcaption.
- The image and the link has an aria-describedby attribute that uses the id.
Also view the front and confirm that the attributes are correct.

Please test the linked image with a screen reader and confirm that the caption is announced when you navigate to the link.

On one block, add an alt text and confirm that the ID and aria-describedby attributes are not present.
On the second block, (which does not have an alt text) remove the caption and confirm that the ID and aria-describedby attributes are not present.


## Screenshots or screencast
